### PR TITLE
Added base64 encode/decode methods to string fixture to work with base64 (representation of) strings

### PR DESCRIPTION
**Pull request by tcnh**
2 convenience methods. Useful f.e. when you need to create auth headers at runtime in a test

### CODE REVIEW

**Comment by fhoeben** [comment]
Are these better of in StringFixture, or should the corresponding methods in Base64Fixture be exposed?

**Comment by tcnh** [comment]
I like them in String Fixture, but that's personal preference. I thought this might prevent confusion (I'd expect that I could send any Object to an encode method on Base64Fixture, instead of just Strings)

It's also more convenient in my practice, because I set StringFixture as a Library by default. I have no need for Base64Fixture in most projects.

**Comment by fhoeben** [comment]
Currently Base64Fixture only supports encoding to/from files in the files section

**Comment by fhoeben** [comment]
Base64 encoding of strings is not the most commonplace requirement I expect, in general that's more something I've encountered for binary data.
Can you elaborate on how this fits into a test, what is the en/decoded data used for? As input to other fixtures?

**Comment by tcnh** [comment]
Mostly basic auth headers with http test. Alternatively we could add a method `set basic authorization header for user | x | and password | y |` in http test.

i.e. a credential comes from another service or from a system property on the build server. We need to translate these values to a base64 string to set as the Authorization header value in http test.

I have never needed to decode base64 strings in a test, but I thought it to be logical to support both ways and not just one.

**Comment by tcnh** [comment]
Method is now in http test. Encode/decode removed from string fixture

**Comment by fhoeben** [comment]
HsacAcceptanceTests.SlimTests.UtilityFixtures.StringFixtureTest now fails...

File: wiki/FitNesseRoot/HsacExamples/SlimTests/UtilityFixtures/StringFixtureTest.wiki [hunk 1]
```diff
@@ -6,19 +6,19 @@ It is intended to be used is as a [[library][.FitNesse.UserGuide.WritingAcceptan
 
 This page demonstrates the features of the class (!-nl.hsac.fitnesse.fixture.slim.StringFixture-!) and therefore uses it on its own in a script table (instead of inside a script using a different fixture).
 
-|script|string fixture                                                                                    |
```
Comment (fhoeben) [inline wiki/FitNesseRoot/HsacExamples/SlimTests/UtilityFixtures/StringFixtureTest.wiki]: Can you revert this entire file to its previous, formatted, state?

File: src/main/java/nl/hsac/fitnesse/fixture/slim/StringFixture.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import nl.hsac.fitnesse.fixture.util.LineEndingHelper;
 import org.apache.commons.lang.StringUtils;
 
+import java.util.Base64;
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/StringFixture.java]: Import can be removed

File: src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java [hunk 3]
```diff
@@ -749,6 +750,12 @@ public void setContentType(String aContentType) {
         contentType = aContentType;
     }
 
+    public void setBasicAuthorizationHeaderForUserAndPassword(String user, String password) {
+        String credential = user + ":" + password;
+        String base64credential = Base64.getEncoder().encodeToString(credential.getBytes());
```
Comment (fhoeben) [inline src/main/java/nl/hsac/fitnesse/fixture/slim/HttpTest.java]: Please add an explicit encoding (using 'ISO-8859-1') when converting the string to bytes.

**Pull request merged**